src/dra/7E4BC.c

```diff
@@ -893,7 +893,94 @@ void EntityHitByIce(Entity* self) {
 INCLUDE_ASM("dra/nonmatchings/7E4BC", EntityTransparentWhiteCircle);
 
 // pink effect on player
-INCLUDE_ASM("dra/nonmatchings/7E4BC", EntityPlayerPinkEffect);
+void EntityPlayerPinkEffect(Entity* self) {
+    s16 paramsHi = (self->params & 0x7F00) >> 8;
+    Unkstruct_800ADEF0* data_idx = (Unkstruct_800ADEF0*)&D_800ADEF0[paramsHi];
+    u32 temp2;
+
+    switch (self->step) {
+    case 0:
+        self->flags = FLAG_UNK_04000000 | FLAG_UNK_40000 | FLAG_UNK_20000 |
+                      FLAG_UNK_10000;
+        self->ext.timer.t = data_idx->unk0[0];
+        if (data_idx->unk18 != 0x83) {
+            PlaySfx(0x668);
+        }
+        if (data_idx->unk18 >= 128) {
+            func_8010E168(true, 64);
+        } else {
+            func_800FF0F4(data_idx->unk18);
+        }
+        self->step += 1;
+        break;
+    case 1:
+        if (--self->ext.timer.t != 0) {
+            return;
+        }
+        if (data_idx->pad2[self->ext.factory.unk7E]) {
+            CreateEntFactoryFromEntity(
+                self,
+                // factory changed
+                (data_idx->pad2[self->ext.factory.unk7E] +
+                 (data_idx->unk16[self->ext.factory.unk7E] << 16)),
+                0);
+            if (data_idx->pad2[self->ext.factory.unk7E] == 0x28) {
+                PlaySfx(0x67D);
+            }
+        }
+
+        self->ext.factory.unk7E++;
+        self->ext.timer.t = data_idx->unk0[self->ext.factory.unk7E];
+        if (self->ext.timer.t == 0xFF) {
+            temp2 = data_idx->unk18;
+            switch (temp2) {
+            case 0x83:
+                if (PLAYER.step == 0xB) {
```

Comment:
```suggestion
                if (PLAYER.step == Player_StatusStone) {
```
Most of the `PLAYER.step` are documented by an enum. `0x0B` should be `Player_StatusStone`!